solidity/test/bridge/Bridge.Deposit.test.ts

```diff
@@ -0,0 +1,2745 @@
+                let tx: ContractTransaction
+
+                before(async () => {
+                  await createSnapshot()
+
+                  tx = await bridge.revealDeposit(P2SHFundingTx, reveal)
+                })
+
+                after(async () => {
+                  await restoreSnapshot()
+                })
+
+                it("should store proper deposit data", async () => {
+                  // Deposit key is keccak256(fundingTxHash | fundingOutputIndex).
+                  const depositKey = ethers.utils.solidityKeccak256(
+                    ["bytes32", "uint32"],
+                    [
+                      "0x17350f81cdb61cd8d7014ad1507d4af8d032b75812cf88d2c636c1c022991af2",
+                      reveal.fundingOutputIndex,
+                    ]
+                  )
+
+                  const deposit = await bridge.deposits(depositKey)
+
+                  // Depositor address, same as in `reveal.depositor`.
+                  expect(deposit.depositor).to.be.equal(
+                    "0x934B98637cA318a4D6E7CA6ffd1690b8e77df637"
+                  )
+                  // Deposit amount in satoshi. In this case it's 10000 satoshi
+                  // because the P2SH deposit transaction set this value for the
+                  // funding output.
+                  expect(deposit.amount).to.be.equal(10000)
+                  // Revealed time should be set.
+                  expect(deposit.revealedAt).to.be.equal(await lastBlockTime())
+                  // Deposit vault, same as in `reveal.vault`.
+                  expect(deposit.vault).to.be.equal(
+                    "0x594cfd89700040163727828AE20B52099C58F02C"
+                  )
+                  // Treasury fee should be computed according to the current
+                  // value of the `depositTreasuryFeeDivisor`.
+                  expect(deposit.treasuryFee).to.be.equal(5)
+                  // Swept time should be unset.
+                  expect(deposit.sweptAt).to.be.equal(0)
+                })
+
+                it("should emit DepositRevealed event", async () => {
+                  await expect(tx)
+                    .to.emit(bridge, "DepositRevealed")
+                    .withArgs(
+                      "0x17350f81cdb61cd8d7014ad1507d4af8d032b75812cf88d2c636c1c022991af2",
+                      reveal.fundingOutputIndex,
+                      "0x934B98637cA318a4D6E7CA6ffd1690b8e77df637",
+                      10000,
+                      "0xf9f0c90d00039523",
+                      "0x8db50eb52063ea9d98b3eac91489a90f738986f6",
+                      "0x28e081f285138ccbe389c1eb8985716230129f89",
+                      "0x60bcea61",
+                      reveal.vault
+                    )
+                })
+              })
+
+              context("when deposit is not routed to a vault", () => {
+                let tx: ContractTransaction
+                let nonRoutedReveal
```

Comment:
Would setting a type make sense here?
```
let nonRoutedReveal: DepositRevealInfoStruct
```

Imported from 
```
import { DepositRevealInfoStruct } from "../../typechain/Bridge"
```

It occurs multiple times in the files.

Reply:
See https://github.com/keep-network/tbtc-v2/pull/270/commits/058ce527345ff8e3bf710a025997203f830c14b6

Reply:
There is still one more left:
```
let nonTrustedVaultReveal
```